### PR TITLE
Added Open Source Contributions with Git Section

### LINKED ISSUE #37: shift Github info from PC guidelines to resources repo

### PR DESCRIPTION
**Pull request by aditya-shirwatkar**
Added Open Source Contributions with Git Section in Software Skills Readme, Also added Bullet list Format in Linux Section due to OCD:smile:

### CODE REVIEW

**Comment by akshaykulkarni07** [comment]
Please also open a PR in pc_guidelines to remove the Git part from it.

**Comment by aditya-shirwatkar** [comment]
Ok,:+1:  I'll create a separate pull request on PC_guidelines and remove this part

**Comment by take2rohit** [comment]
Change landing page readme and try to include the name git also. 

I would recommend to create a new folder named `git-github`. We can add courses on git also like

[coursera course by Google](https://www.coursera.org/learn/introduction-git-github)

And one video lecture from `missing semester`
https://youtu.be/2sjqTHE0zok

You can also include my lecture video given on git to first year 😛


If @akshaykvnit @aayush-fadia @aditya-shirwatkar @SharathRaparthy have some other courses or resources to recommend then do tell us in this thread. We will review and add it 😊

**Comment by akshaykulkarni07** [comment]
I guess Coursera course is already there in the files changed. @take2rohit you can add the missing semester lecture link I think to this itself. Unless there is extensive stuff to be added, let's keep it together for now. You can of course open a new PR to separate  and organize into folders, once you find enough things you want to add.

File: software/README.md [hunk 1]
```diff
@@ -2,6 +2,36 @@
```
Comment (akshaykulkarni07) [inline software/README.md:2]: add H1 sized "Software" (for consistency with other folders)

File: software/README.md [hunk 2]
```diff
@@ -2,6 +2,36 @@
...
+2. Create a new fork and do all changes there.
+3. Create a Pull Request and repeat above steps.
+4. Thats it!
+
+### Courses
+- [Introduction to Git and GitHub](https://www.coursera.org/learn/introduction-git-github#syllabus) 
+This is a highly recommended course for anyone new to Git. It also covers a few advance topics. 
+
+This is an oversimplified way of contributing
+For more advanced usage we can use branches and commands like adding upstream url and doing `git fetch`. Please read online about these concepts. Some great explanations can be found [here](https://gist.github.com/Chaser324/ce0505fbed06b947d962) and [here](https://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository). They're a must read for everyone.
```
Comment (akshaykulkarni07) [inline software/README.md]: put a full stop between ".... way of contributing" and "For more advanced usage...".
put a comma after usage ("For more advanced usage, we can use branches and ......")
Change sentence "Please read online about these concepts" to "Please read about these concepts online".

**Comment by take2rohit** [comment]
once these are done make sure to do changes in landing page readme of resources and paper notes

1. In point 4 of `Guidelines for contributing` (https://github.com/IvLabs/resources)
2.  In point 4 of `Rules of appending papers`  (https://github.com/IvLabs/ResearchPaperNotes)

After all these changes, this PR will be ready for merge

File: software/README.md (hunk 1 above)
Comment (aditya-shirwatkar) [reply software/README.md:2]: A single "#" means H1 right ?, it is already there, I didn't understand the issue :sweat_smile:

File: software/README.md (hunk 2 above)
Comment (aditya-shirwatkar) [reply software/README.md]: Added the changes

File: software/README.md (hunk 1 above)
Comment (akshaykulkarni07) [reply software/README.md:2]: Oh okay, it was hidden since it wasn't the changed part. Sorry. It's correct.

**Comment by aditya-shirwatkar** [comment]
> once these are done make sure to do changes in landing page readme of resources and paper notes
> 
> 1. In point 4 of `Guidelines for contributing` (https://github.com/IvLabs/resources)
> 2. In point 4 of `Rules of appending papers`  (https://github.com/IvLabs/ResearchPaperNotes)
> 
> After all these changes, this PR will be ready for merge

What changes ?, We are keeping git and software together for now, right :sweat_smile:?

File: software/README.md [hunk 3]
```diff
@@ -33,5 +33,5 @@
 - [Introduction to Git and GitHub](https://www.coursera.org/learn/introduction-git-github#syllabus) 
 This is a highly recommended course for anyone new to Git. It also covers a few advance topics.
```
Comment (akshaykulkarni07) [inline software/README.md]: .....covers a few _advanced_ topics.

**Comment by akshaykulkarni07** [comment]
@aditya-shirwatkar He means that in the landing page (main README), the link to open source contributions should now point to the software README instead of earlier pc_guidelines repo readme.

**Comment by aditya-shirwatkar** [comment]
> @aditya-shirwatkar He means that in the landing page (main README), the link to open source contributions should now point to the software README instead of earlier pc_guidelines repo readme.

Done

**Review by akshaykulkarni07 — APPROVED** [review]

**Comment by aditya-shirwatkar** [comment]
@take2rohit Created Pull request 25 in @Ivlabs/ResearchPaperNotes
To update point 4 Hyperlink in Rules of Appending Papers

**Comment by take2rohit** [comment]
Nice work @aditya-shirwatkar I did some changes in ee40e8a 
merging this pull request in case of any issues feel free to reopen this PR.

**Pull request merged**